Agents/AQMeshInputAgent/AQMeshInputAgent/src/main/java/uk/ac/cam/cares/jps/agent/aqmesh/AQMeshInputAgent.java

```diff
@@ -394,6 +396,49 @@ private TimeSeries<OffsetDateTime> pruneTimeSeries(TimeSeries<OffsetDateTime> ti
         return new TimeSeries<>(newTimes, timeSeries.getDataIRIs(), newValues);
     }
 
+    public JSONArray calculateAndAddScaledMeasurementsForParticle(JSONArray readings) {
+        List<String> list = Arrays.asList("pm1", "pm2_5", "pm4", "pm10", "pm_tpc", "pm_total");
+        for (int i = 0; i < readings.length(); i++) {
+            JSONObject currentEntry = readings.getJSONObject(i);
+            for (int j = 0; j < list.size(); j++) {
+                String key = list.get(j);
+                Object prescaled = currentEntry.get(key + "_prescale");
+                Object slope = currentEntry.get(key + "_slope");
+                Object offset = currentEntry.get(key + "_offset");
```

Comment:
I'd avoid using Object unless it is absolutely necessary

Reply:
The purpose of using Object is tied in with the next line 408 to check for NULL using JSOBObject.NULL. Because if I try to use Double and currentEntry.getDouble, assuming the value is a null, it will throw an exception.